foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -119,6 +137,123 @@ def get_active_volumes() -> Set[str]:
     return volumes
 
 
+##
+# Job Prioritization
+##
+
+
+def prioritize_salmon_jobs(jobs: List) -> List:
+    """Prioritizes salmon experiments based on how close to completion they are.
+
+    This is because salmon experiments have a final processing step
+    that must be performed on all the samples in the experiment, so if
+    9/10 samples in an experiment are processed then they can't
+    actually be used until that last sample is processed.
+    """
+    # The strategy for doing so is to build a mapping between every
+    # salmon job in `jobs` to a priority. This priority will be what
+    # percentage of the samples in this experiment have been
+    # processed. Once we have that mapping we can sort those jobs by
+    # that priority and move them to the front of the list.
+    prioritized_jobs = []
+    for job in jobs:
+        # Salmon jobs are specifc to one sample.
+        sample = job.get_samples().pop()
+
+        # Skip jobs that aren't for Salmon. Handle both ProcessorJobs and DownloaderJobs.
+        if type(job) is ProcessorJob and job.pipeline_applied != ProcessorPipeline.SALMON.value:
+            continue
+        elif type(job) is DownloaderJob:
+            is_salmon_sample = False
+            for original_file in sample.original_files.all():
+                if original_file.filename[-5:].upper() == "FASTQ" \
+                or original_file.filename[-8:].upper() == "FASTQ.GZ" \
+                or original_file.filename[-2:].upper() == "FQ" \
+                or original_file.filename[-3:].upper() == "SRA" \
+                or original_file.filename[-5:].upper() == "FQ.GZ":
```

Comment:
Are these based on real data, or just memory? I feel like some (compressed?) file formats might be missing here.

Reply:
I got these out of `common/job_lookup.py` because that's what we use to decide if a file is RNA-seq there.

Reply:
Roger roger. In that case, can it be functionalized?